src/cfg/cfg-traversal.h

```diff
@@ -45,7 +45,13 @@ struct CFGWalker : public ControlFlowWalker<SubType, VisitorType> {
     std::vector<BasicBlock*> out, in;
   };
 
-  BasicBlock* entry; // the entry block
+  // The entry block at the function's start. This always exists.
+  BasicBlock* entry;
+
+  // The exit block at the end where control flow leaves the function. This may
```

Comment:
Maybe "reaches the end" rather than "leaves" to make it clearer that `return` doesn't count.

Reply:
Ah, that is better, thanks.